resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/security/HadoopFSCredentialProvider.scala

```diff
@@ -81,8 +90,15 @@ private[security] class HadoopFSCredentialProvider
     sparkConf.get(PRINCIPAL).flatMap { renewer =>
       val creds = new Credentials()
       hadoopFSsToAccess(hadoopConf, sparkConf).foreach { dst =>
-        val dstFs = dst.getFileSystem(hadoopConf)
-        dstFs.addDelegationTokens(renewer, creds)
+        try {
+          val dstFs = dst.getFileSystem(hadoopConf)
+          dstFs.addDelegationTokens(renewer, creds)
+        } catch {
+          case e: StandbyException =>
+            logWarning(s"Namenode ${dst} is in state standby", e)
+          case e: RemoteException =>
```

Comment:
I'd suggested adding a handler for UnknownHostException too, but now I think that could hide problems with client config. Best to leave as is.